### PR TITLE
Renamed 'vectordb' to 'vector_db', following snake case convention

### LINKED ISSUE #734: Follow snake case convention in folder vectordb
### 🐛 Describe the bug

Located at  embedchain/config/vectordb

It should be vector_db.

This task will involve changing the relevant imports everywhere and making sure that tests pass and nothing breaks.

### PR DESCRIPTION
**Pull request by PatelKeviin**
## Description
Implemented snake case conventions in the following project structure:
- `embedchain/config/vector_db`
- `embedchain/vector_db`

Updated references everywhere, including configurational yaml files, documentation and testing files.

Fixes #734 (issue)

## Type of change

- [x] Refactor (does not change functionality, e.g. code style improvements, linting)
- [x] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [x] closes #734  (Replace xxxx with the GitHub issue number)
- [x] Made sure Checks passed

### CODE REVIEW

File: configs/chroma.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ llm:
     top_p: 1
     stream: false
 
-vectordb:
+vector_db:
```
Comment (deshraj) [inline configs/chroma.yaml:15]: Hey @PatelKeviin, thanks for the contribution. We only want to change the yaml file structure from `vectordb` to `vector_db` and not inside the yaml file configuration. Can you please revert these changes in the yaml config?

Comment (PatelKeviin) [reply configs/chroma.yaml]: @deshraj thanks for your feedback. Please review my changes - I've reverted the yaml config file updates as advised, as well as reverted back the documentation reference updates to align.

**Comment by deshraj** [comment]
@PatelKeviin sorry for the delay. Can you please resolve the conflicts so that we can do a final review?

**Comment by taranjeet** [comment]
Thanks for your contribution @PatelKeviin .
As there are so many conflicts on this PR, it would make a lot of sense to start afresh and open a new PR.
Feel free to tag us if you run into any issues.
